### PR TITLE
rana: init at 0.5.4

### PR DESCRIPTION
**Pull request by jigglycrumb**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix
Comment (ocfox) [inline pkgs/top-level/all-packages.nix:26228]:
```suggestion
```

File: pkgs/by-name/ra/rana/default.nix
Comment (ocfox) [inline pkgs/by-name/ra/rana/default.nix:1]: pkgs/by-name/ra/rana/package.nix

**Comment by ocfox** [comment]
You can test it by run 'nix run .#rana' in your branch.

**Comment by ocfox** [comment]
Not needed to callPackage if use by-name, remove it.

**Comment by jigglycrumb** [comment]
> nix run .#rana

Done, this works for me :+1:

**Comment by ocfox** [comment]
And don't forget to remove useless commits, only need `add maintainer` and `init`.

**Comment by jigglycrumb** [comment]
@ocfox Thank you for your help so far, this was a great first experience with this repo! I will remove the useless commits tomorrow - I have to jump out now for today, and I need some peer help to actually do it (not experienced with rebase and scared to break things :sweat_smile:)

Looking forward to get this merged :grin:

**Comment by ocfox** [comment]
It's okay. I'm glad to help you. Be assured to use force push.

File: pkgs/by-name/ra/rana/package.nix
Comment (ocfox) [inline pkgs/by-name/ra/rana/package.nix:9]:
```suggestion
    repo = "rana";
```

File: pkgs/by-name/ra/rana/package.nix
Comment (ocfox) [inline pkgs/by-name/ra/rana/package.nix:21]:
```suggestion
    mainProgram = "rana";
  };
```

File: pkgs/by-name/ra/rana/package.nix
Comment (ocfox) [inline pkgs/by-name/ra/rana/package.nix:1]: Try use `nixfmt-rfc-style` format.

**Review by ocfox — CHANGES_REQUESTED** [review]

**Comment by ocfox** [comment]
Result of `nixpkgs-review pr 325911` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rana</li>
  </ul>
</details>

**Review by ocfox — APPROVED** [review]

**Comment by jigglycrumb** [comment]
@ocfox Should I remove the last commit again or is it fine like this? I guess you'd have to approve again if I do this.

**Comment by ocfox** [comment]
I think no problem to keep it.

**Comment by jigglycrumb** [comment]
Hello :) Just wondering if there is anything left that I should do? I don't want to block this without knowing 😅

**Comment by marcusramberg** [comment]
> Hello :) Just wondering if there is anything left that I should do? I don't want to block this without knowing 😅 

Please squash your fix commit into the init one. Otherwise lgtm.

File: pkgs/by-name/ra/rana/package.nix
Comment (Aleksanaa) [inline pkgs/by-name/ra/rana/package.nix:20]: Better to avoid `with lib`.

**Comment by jigglycrumb** [comment]
Any news? :pray:

**Comment by jigglycrumb** [comment]
Bump

**Comment by Aleksanaa** [comment]
@ofborg build rana

**Comment by Aleksanaa** [comment]
I'm very sorry for ignorance. Just rebuilding it to avoid rust 1.80 breakage

**Comment by jigglycrumb** [comment]
No problem, thanks for merging!

**Pull request merged**
